setup.py

```diff
@@ -0,0 +1,833 @@
+                    print(f"One or more canbus_uuids found in the log.")
+                else:
+                    print("No canbus_uuids found in the log.")
+
+            return found
+    except FileNotFoundError:
+        print(f"Log file not found at: {klippy_log_path}")
+        return False
+    except Exception as e:
+        print(f"An error occurred while reading the log file: {str(e)}")
+        return False
+
+
+# Function to find Cartographer serial IDs
+def find_cartographer_serial_id():
+    serial_dir = "/dev/serial/by-id"
+    try:
+        serial_ids = [f for f in os.listdir(serial_dir) if "Cartographer" in f]
+        return os.path.join(serial_dir, serial_ids[0]) if serial_ids else None
+    except FileNotFoundError:
+        print(f"Directory {serial_dir} not found.")
+        return None
+
+
+# Function to create a backup of the configuration file
+def backup_config_file():
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+    backup_file_path = os.path.expanduser(
+        f"~/printer_data/config/printer.cfg.bak_{timestamp}"
+    )
+    shutil.copy(config_file_path, backup_file_path)
+    print(f"Backup of printer.cfg created at {backup_file_path}")
+
+
+# Function to add configuration lines based on probe type
+def add_probe_config(probe_type):
+    if debug:
+        # Heading with consistent formatting
+        print("\n" + "=" * 60)
+        print(f" Debugging Output for add_probe_config ".center(60, "="))
+        print("=" * 60 + "\n")
+        print(f"Adding configuration for probe type: touch")
+
+        print("Configuring settings for touch probe...")
+
+    # backup_config_file()  # Backup the config file before editing
+
+    with open(config_file_path, "r") as config_file:
+        lines = config_file.readlines()
+
+    scanner_exists = any(line.strip().startswith("[scanner]") for line in lines)
+    cartographer_exists = any(
+        line.strip().startswith("[cartographer]") for line in lines
+    )
+
+    # Adding [scanner] section for touch probe type
+    if not scanner_exists and not cartographer_exists:
+        canbus_uuid = get_canbus_uuid()
+
+        serial_entry = "#serial: /dev/serial/by-id/ # CHANGE ME FOR USB"
+        canbus_entry = "#canbus_uuid:  # CHANGE ME FOR CANBUS"
+
+        if canbus_uuid and not check_canbus_in_log(canbus_uuid):
+            canbus_entry = f"canbus_uuid: {canbus_uuid}"
+        else:
+            serial_id = find_cartographer_serial_id()
+            if serial_id:
+                serial_entry = f"serial: {serial_id}"
+                canbus_entry = "#canbus_uuid: # CHANGE ME FOR CANBUS"
+
+        # Scanner config lines
+        config_lines = f"""
+[scanner]
+{serial_entry}            
+{canbus_entry}
+#
+#   Visit the link below for help finding your device ID
+#   https://docs.cartographer3d.com/cartographer-probe/installation-and-setup/classic-installation/klipper-setup#finding-the-serial-or-uuid
+#        
+x_offset: 0              
```

Comment:
Can we somehow get the current? Otherwise I think they should be commented out, so that it triggers an error?

Reply:
carto is 99% alwats x: 0 offset

Reply:
Yeah, sorry, was thinking y, primarily :)